rust-toolchain.toml

```diff
@@ -1,3 +1,3 @@
 [toolchain]
-channel = "1.66"
+channel = "1.70"
```

Comment:
If we bump this, do we need it bumped anywhere else in Omicron/Propolis?
I don't think so, as the downstairs itself is standalone, but not sure about any
other consumers of the upstairs

Reply:
I'm not sure, leaning towards yes - I can't see how (for example) Propolis would be able to compile this in as a library without support for `Arc::into_inner`. We'll see when that rev gets bumped I guess :)

Reply:
We might want to try to bump propolis and see if it compiles.  If there is some entanglement
and it turns out to be a huge amount of work, we might consider holding this fix till we can
get propolis ready for us.  I'm sure we will have other changes we want in Crucible and I want
to be able to update omicron/propolis frequently over the next few weeks.

Reply:
i was compiling latest propolis against this commit during my perf tests fwiw